libmscore/realizedharmony.cpp

```diff
@@ -0,0 +1,551 @@
+                        else if (literal)
+                              ret.insert(5 + RANK_MULT*RANK_OMIT, tpcInterval(rootTpc, 11, 0));     //maj11
+                        omit |= 1 << 11;
+                        }
+                  // FALLTHROUGH
+            case 9:
+                  if (!(omit & (1 << 9))) {
+                        ret.insert(2 + RANK_MULT*RANK_9TH, tpcInterval(rootTpc, 9, 0));     //maj9
+                        omit |= 1 << 9;
+                        }
+                  // FALLTHROUGH
+            case 7:
+                  if (!(omit & (1 << 7))) {
+                        if (quality == "major")
+                              ret.insert(11 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, 0));
+                        else if (quality == "diminished")
+                              ret.insert(9 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, -2));
+                        else //dominant or augmented or minor
+                              ret.insert(10 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, -1));
+                        }
+                  break;
+            case 6:
+                  if (!(omit & (1 << 6))) {
+                        ret.insert(9 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 6, 0));     //maj6
+                        omit |= 1 << 13; //no need to add/alter 6 chords
+                        }
+                  break;
+            case 5:
+                  //omitted third already
+                  break;
+            case 4:
+                  if (!(omit & (1 << 4)))
+                        ret.insert(5 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 4, 0));     //p4
+                  break;
+            case 2:
+                  if (!(omit & (1 << 2)))
+                        ret.insert(2 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 2, 0));     //maj2
+                  omit |= 1 << 9; //make sure we don't add altered 9 when theres a 2
+                  break;
+            case 69:
+                  ret.insert(9 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 6, 0));     //maj6
+                  ret.insert(2 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 9, 0));     //maj6
+                  omit = ~0; //no additions/alterations for a 69 chord
+                  break;
+            default:
+                  break;
+            }
+
+      Harmony* next = _harmony->findNext();
+      if (!_literal && next) {
+            //jazz interpretation
+            QString qNext = next->parsedForm()->quality();
+            //pitch from current to next harmony normalized to a range between 0 and 12
+            //add PITCH_DELTA_OCTAVE before modulo so that we can ensure arithmetic mod rather than computer mod
+            //int keyTpc = int(next->staff()->key(next->tick())) + 14; //tpc of key (ex. F# major would be Tpc::F_S)
+            //int keyTpcMinor = keyTpc + 3;
+            int pitchBetween = (tpc2pitch(next->rootTpc()) + PITCH_DELTA_OCTAVE - tpc2pitch(rootTpc)) % PITCH_DELTA_OCTAVE;
```

Comment:
next->rootTpc() can be invalid, causing this to crash (well, assertion failure in tpc2pitch).  There is a valid use case where this can happen, chord symbols specified as for example "/G", meaning, repeat the previous chord but with a G in the bass.  Probably these just should be skipped, since we don't really do anything special with bass information.  And other invalid chords definitely need to be skipped.  Maybe findNext() should just skip any Harmony objects with an invalid rootTpc.

Reply:
Either that, or we check tpcIsValid(next->rootTpc()) in the "if" statement above.  That would just skip the block of code is the next chord is something we don't recognize, rather than continuing to look for something we do recognize.

The effect of this chunk of code is subtle.  It only applies to jazz interpretation, and it has to do with adding appropriate extensions to a chord based on what the next chord is.

Reply:
I'm added tpcIsValid(next->rootTpc())